programs/clearing_house/src/lib.rs

```diff
@@ -1490,6 +1491,90 @@ pub mod clearing_house {
         Ok(())
     }
 
+    #[access_control(
+        exchange_not_paused(&ctx.accounts.state)
+    )]
+    pub fn update_amm(ctx: Context<UpdateAMM>, market_indexes: [u64; 5]) -> Result<()> {
+        // ~60k compute units
+
+        let remaining_accounts_iter = &mut ctx.remaining_accounts.iter().peekable();
+        let clock = Clock::get()?;
+        let clock_slot = clock.slot;
+        let now = clock.unix_timestamp;
+        // let user = &mut load_mut(&ctx.accounts.user)?;
+
+        let mut oracle_map = OracleMap::load(remaining_accounts_iter, clock_slot)?;
+        let _bank_map = BankMap::load(&WritableMarkets::new(), remaining_accounts_iter)?;
+
+        // let market_map = MarketMap::load(
+        //     &get_writable_markets_for_user_positions(&user.positions),
+        //     &MarketOracles::new(), // oracles validated in calculate liquidation status
+        //     remaining_accounts_iter,
+        // )?;
+        let market_map = MarketMap::load(
+            &get_writable_markets_list(market_indexes),
+            // &get_market_oracles(market_index, &ctx.accounts.oracle),
+            &MarketOracles::new(),
+            remaining_accounts_iter,
+        )?;
+
+        // let oracle = &ctx.accounts.oracle;
+
+        for market_index in market_indexes.iter() {
+            if *market_index == 100 {
+                continue; //todo
+            }
+
+            // let mark_price_before: u128;
+            // let _oracle_mark_spread_pct_before: i128;
+            // let is_oracle_valid: bool;
+            // let oracle_price: i128;
+
+            let market = &mut market_map.get_ref_mut(market_index)?;
+            let oracle_price_data = oracle_map.get_price_data(&market.amm.oracle)?;
+            // let oracle_price_data = &market.amm.get_oracle_price(oracle, clock_slot)?;
+
+            let prepeg_budget = repeg::calculate_fee_pool(market)?;
+            let state = &ctx.accounts.state;
+
+            let is_oracle_valid = amm::is_oracle_valid(
+                &market.amm,
+                oracle_price_data,
+                &state.oracle_guard_rails.validity,
+            )?;
+
+            controller::repeg::prepeg(
```

Comment:
do you want to stick with `prepeg` naming in the long run? not sure if it's an intuitive enough name

Reply:
can just call it repeg if thats better,

re-"base" is another name we could use